syntactic-paths/src/main/java/com/palantir/util/syntacticpath/Path.java

```diff
@@ -113,14 +103,10 @@ private static List<String> checkSegments(List<String> segments) {
     }
 
     private Path normalizeInternal() {
-        LinkedList<String> normalSegments = Lists.newLinkedList();
+        Deque<String> normalSegments = new ArrayDeque<>(segments.size());
         for (String segment : segments) {
             if (segment.equals(BACKWARDS_PATH)) {
-                if (!normalSegments.isEmpty()) {
-                    normalSegments.removeLast();
-                } else {
-                    // Nothing to do.
-                }
```

Comment:
This `LinkedList` was erroring as [`JdkObsolete`](https://errorprone.info/bugpattern/JdkObsolete) so went with `ArrayDeque` and simplified the code a bit

See also:
- https://stuartmarks.wordpress.com/2015/12/18/some-java-list-benchmarks/
- https://github.com/palantir/atlasdb/pull/6187
- https://github.com/palantir/atlasdb/pull/5682